client/packages/coldchain/src/Equipment/ListView/AddFromScannerButton.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useEffect, useRef } from 'react';
+import {
+  useTranslation,
+  useBarcodeScannerContext,
+  CircularProgress,
+  ScanIcon,
+  ScanResult,
+  ButtonWithIcon,
+  useNotification,
+  useRegisterActions,
+  Box,
+  useNavigate,
+  useDisabledNotification,
+  RouteBuilder,
+} from '@openmsupply-client/common';
+import { AssetFragment } from '../api';
+import { AppRoute } from '@openmsupply-client/config';
+
+export const AddFromScannerButtonComponent = ({
+  assets,
+}: {
+  assets: AssetFragment[];
+}) => {
+  const t = useTranslation('coldchain');
+  const { isConnected, isEnabled, isScanning, startScanning, stopScan } =
+    useBarcodeScannerContext();
+  const { error } = useNotification();
+  const buttonRef = useRef<HTMLButtonElement>(null);
+  const navigate = useNavigate();
+  const { DisabledNotification, show } = useDisabledNotification({
+    title: t('error.unable-to-scan'),
+    message: t('error.scanner-not-connected'),
+  });
+  const equipmentRoute = RouteBuilder.create(AppRoute.Coldchain).addPart(
+    AppRoute.Equipment
+  );
+
+  const handleScanResult = async (result: ScanResult) => {
+    if (!!result.content) {
+      const { content } = result;
+      const id = content;
+      if (assets.some(asset => asset.id === id)) {
```

Comment:
Will this have a problem with pagination? Should we just try to navigate to the id and return and error if it's invalid?

Reply:
well spotted! yes - it will have a pagination issue. that's a shame, it's pretty clunky navigating to the detail view and popping an error there. 

I've converted to an api fetch to query and redirect if valid.